File: packages/javascript/lib/date/dateToYYYYMMDD.ts
export function dateToYYYYMMDD(date: Date, separator = '-'): string {
	const month: number = date.getMonth() + 1;
	const day: number = date.getDate();

	return `${date.getFullYear()}${separator}${month < 10 ? `0${month}` : month}${separator}${
		day < 10 ? `0${day}` : day
	}`;
}

File: packages/python/qsu/date/dateToYYYYMMDD.py
from datetime import date as _date
from datetime import datetime


def dateToYYYYMMDD(date, separator: str = '-') -> str:
	if isinstance(date, datetime):
		year = date.year
		month = date.month
		day = date.day
	elif isinstance(date, _date):
		year = date.year
		month = date.month
		day = date.day
	else:
		raise TypeError('date must be a datetime or date instance')

	month_str = f'0{month}' if month < 10 else f'{month}'
	day_str = f'0{day}' if day < 10 else f'{day}'

	return f'{year}{separator}{month_str}{separator}{day_str}'
